File: app/components/folio/console/tiptap/simple_form_wrap/word_count_component.js
window.Folio.Stimulus.register('f-c-tiptap-simple-form-wrap-word-count', class extends window.Stimulus.Controller {
  static targets = ['wordsCount', 'charactersCount']

  static values = {
    attributeName: String
  }

  updateWordCount (e) {
    if (!e?.detail?.wordCount) return
    this.updateCounts(e.detail.wordCount)
  }

  updateCounts (wordCount) {
    this.wordsCountTargets.forEach((target) => { target.innerText = wordCount.formattedWords })
    this.charactersCountTargets.forEach((target) => { target.innerText = wordCount.formattedCharacters })
  }
})

File: app/components/folio/console/tiptap/simple_form_wrap/word_count_component.rb
# frozen_string_literal: true

class Folio::Console::Tiptap::SimpleFormWrap::WordCountComponent < Folio::Console::ApplicationComponent
  def initialize(attribute_name:, data: nil)
    @attribute_name = attribute_name
    @data = data || {}
  end

  def controller_data
    stimulus_merge(@data,
                  stimulus_controller("f-c-tiptap-simple-form-wrap-word-count",
                                      values: {
                                        attribute_name: @attribute_name,
                                      },
                                      action: {
                                        "f-c-tiptap-simple-form-wrap:updateWordCount" => "updateWordCount",
                                      }))
  end
end
